src/main/java/de/zalando/tip/zalenium/proxy/DockerSeleniumStarterRemoteProxy.java

```diff
@@ -493,6 +493,16 @@ private boolean validateAmountOfDockerSeleniumContainers() {
         Method adapted from https://gist.github.com/vorburger/3429822
      */
     private int findFreePortInRange(int lowerBoundary, int upperBoundary) {
+        /*
+            If the list size is this big (~9800), it means that almost all ports have been used, but
+            probably many have been released already. The list is cleared so ports can be reused.
+            If by any chance one of the first allocated ports is still used, it will be skipped by the
+            existing validation.
+         */
+        if (allocatedPorts.size() > (upperBoundary - lowerBoundary - 200)) {
+            allocatedPorts.clear();
+            LOGGER.log(Level.INFO, LOGGING_PREFIX + "Cleaning allocated ports list.");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)